multimodal-search-pdf/pods/segmenters.py

```diff
@@ -0,0 +1,285 @@
+    def segment(self, text: str, uri: str, buffer: bytes, mime_type: str, *args, **kwargs) -> List[Dict]:
+        """
+        Segements text.
+
+        :param text: the text data
+        :param uri: File name of PDF
+        :type uri: str
+        :param buffer: PDF file in bytes
+        :type buffer: bytes
+        :param mime_type: the type of data
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+
+        chunks = []
+        if mime_type == 'text/plain':
+            chunks.append(dict(text=text, weight=1.0, mime_type='text/plain'))
+        return chunks
+
+
+class TextSegmenterCustomized(BaseSegmenter):
+    """
+    :class:`TextSegmenterCustomized` segments text into lines
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @single(slice_nargs=4)
+    def segment(self, text: str, uri: str, buffer: bytes, mime_type: str, *args, **kwargs) -> List[Dict]:
+        """
+        Segements text.
+
+        :param text: the text data
+        :param uri: File name of PDF
+        :type uri: str
+        :param buffer: PDF file in bytes
+        :type buffer: bytes
+        :param mime_type: the type of data
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+        chunks = []
+        if mime_type == 'text/plain':
+            temp = text.split('\n')
+            for t in temp:
+                chunks.append(dict(text=t, weight=1.0, mime_type='text/plain'))
+        return chunks
+
+
+class ImageSegmenter(BaseSegmenter):
+    """
+    :class:`ImageSegmenter` reads image and stores into chunks.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @single(slice_nargs=3)
+    def segment(self,  uri: str, buffer: bytes, mime_type: str, *args, **kwargs) -> List[Dict]:
+        """
+        Segements image.
+
+        :param uri: File name of image
+        :type uri: str
+        :param buffer: image file in bytes
+        :type buffer: bytes
+        :param mime_type: the type of data
+        :returns: A list of documents with the extracted data
+        :rtype: List[Dict]
+        """
+        chunks=[]
+        if mime_type.startswith('image'):
+            from PIL import Image
+            self.channel_axis: int = -1
+
+            if buffer:
+                raw_img = Image.open(io.BytesIO(buffer))
+            elif uri:
+                raw_img = Image.open(uri)
+            else:
+                raise ValueError('no value found in "buffer" and "uri"')
+            raw_img = raw_img.convert('RGB')
+            img = np.array(raw_img).astype('float32')
+            if self.channel_axis != -1:
+                img = np.moveaxis(img, -1, self.channel_axis)
+
+            chunks = [dict(blob=img, weight=1.0, mime_type='image/png')]
+            return chunks
+        return chunks
+
+
+class MultimodalSegmenter(BaseSegmenter):
```

Comment:
It is better to inherit from the `PDFSegmenter` because they share quite a lot of codes

Reply:
How about removing this class? I intend to keep it just as a backup and not used in the current flow.

Reply:
removed